apps/vast/pyvast_threatbus/pyvast_threatbus.py

```diff
@@ -320,6 +325,22 @@ async def retro_match_vast(
             kwargs["max_events"] = retro_match_max_events
         proc = await vast.export(**kwargs).json(query).exec()
         reported = 0
+        try:
+            # we will likely not hit the deadlock condition described in the
+            # documentation for proc.wait() in combination with PIPEs, because
+            # we do not expect as much data being sent by VAST. Usually, a user
+            # would limit retro matches to a few 10s to hundreds (at max) using
+            # `retro_match_max_events`.
```

Comment:
Yes, we originally wanted to stream results as soon as they become available. But `communicate()` would hinder streaming, as it would block until everything is read into the response and the process exited. So all the startup / teardown times of VAST export would also affect response times.

Reply:
But if I understand correctly, the implementation in this PR *also* hinders streaming? As far as I understand, `asyncio.wait_for()` will wait until `pc.wait()` has completed or the timeout has elapsed, and only afterwards will the while-loop start. Or am I misunderstanding how asyncio works here?

Reply:
No, you are correct. However, there might be some sightings already in the `proc.stdout` buffer. When we'd use `communicate` we'd either get all or none.

Reply:
We discussed in a call to go with `communicate()`.

Reply:
Ok, makes sense.